File: homework/HW2/BadVersion.java
package homework.HW2;

public class BadVersion{

    public int firstBadVersion(int n){
        
        int low = 1;
        int high = n;

        while(low <= high){
            int mid = low + (high - low) / 2;

            if(isBadVersion(mid) == true && isBadVersion(mid - 1) == false){
                return mid;
            }
            else if (isBadVersion(mid) == true){
                high = mid;
            }
            else{
                low = mid + 1;
            }
        }
        return -1;
    }


}

File: homework/HW2/BadVersion.py
class BadVersion:

    def firstBadVersion(n):
        low = 1
        high = n
        while low <= high:
            mid = low + (high - low) / 2

            if(isBadVersion(mid) == True and isBadVersion(mid - 1) == False): return mid
            elif(isBadVersion(mid) == True): high = mid
            else: low = mid + 1
        return -1
    
